packages/plugin-team/src/decrypter.js

```diff
@@ -0,0 +1,64 @@
+/**!
+ *
+ * Copyright (c) 2015-2016 Cisco Systems, Inc. See LICENSE file.
+ * @private
+ */
+
+import {base64} from '@ciscospark/common';
+import {Decrypter} from '@ciscospark/plugin-conversation';
+
+/**
+ * Decrypts a team
+ * @param {Object} team
+ * @param {Encryption~Key|string} key
+ * @param {Object} options
+ * @param {Object} parentObject
+ * @returns {Promise} Resolves with the decrypted team
+ * @private
+ */
+Decrypter.prototype._decryptTeam = function _decryptTeam(team, key, options, parentObject) {
+  let promises = [];
+  if (team.conversations.items) {
+    promises = team.conversations.items.map((item) => this.decryptObject(item, options));
+  }
+
+  const usableKey = key || team.encryptionKeyUrl;
+
+  if (usableKey && team.displayName) {
+    try {
+      const parts = team.displayName.split(`.`);
+      if (parts.length === 5) {
+        JSON.parse(base64.decode(parts[0]));
+
+        // Much like a conversation, a team can be the object of update, add or leave activities.
+        // If this is the case we measure decryption of parent object
+        this.spark.encryption.metrics.startDecryptionMetrics(parentObject || team);
+
+        promises.push(this.decryptProperty(team, `displayName`, usableKey, () => {
+          team.displayName = this.spark.encryption.config.decryptionFailureMessage;
+        }));
+        promises.push(this.decryptProperty(team, `summary`, usableKey, () => {
```

Comment:
decryptProperty no longer accepts the failure handler (in large part because the it never really seemed to work that well). Let's consider doing something more generic; maybe we can add an object of undecryptableKeyPaths to the thing that can't be decrypted and pass the error message up to the client.